include/silo/config/config_exception.h

```diff
@@ -0,0 +1,15 @@
+#ifndef SILO_INCLUDE_SILO_CONFIG_CONFIG_EXCEPTION_H_
+#define SILO_INCLUDE_SILO_CONFIG_CONFIG_EXCEPTION_H_
+
+#include <iostream>
+
+namespace silo {
```

Comment:
Maybe put those in a separate namespace?
```suggestion
namespace silo::config {
```